service/controller/aws/v3/resource_set.go

```diff
@@ -160,16 +161,41 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 		}
 	}
 
+	var chartConfigResource controller.Resource
+	{
+		c := chartconfig.Config{
+			BaseClusterConfig:        *config.BaseClusterConfig,
+			G8sClient:                config.G8sClient,
+			Guest:                    guestClusterService,
+			K8sClient:                config.K8sClient,
+			Logger:                   config.Logger,
+			ProjectName:              config.ProjectName,
+			ToClusterGuestConfigFunc: toClusterGuestConfig,
+		}
+
+		ops, err := chartconfig.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		chartConfigResource, err = toCRUDResource(config.Logger, ops)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
 	resources := []controller.Resource{
 		// Put encryptionKeyResource first because it executes faster than
 		// awsConfigResource and could introduce dependency during cluster
 		// creation.
 		encryptionKeyResource,
 		awsConfigResource,
-		// namespaceResource and chartResource manage resources in the guest
-		// cluster so they should be executed last.
+		// namespace, chart and chartconfig resources manage resources in
+		// guest clusters so they should be executed last.
 		namespaceResource,
 		chartResource,
+		chartResource,
```

Comment:
Don't know much about this, but is it intentional that `chartResource` is double now?

Reply:
It should only appear once. Fixed. Good catch thanks!